File: Design Patterns/basic-patterns-java/memento/player/PlayerMemento.java
package memento.player;

import java.time.LocalDate;

public class PlayerMemento implements Memento {

    private Player player;
    private String name;
    private int point;
    private int level;
    private LocalDate created;


    public PlayerMemento(Player p) {
        this.player = p;
        this.name = p.getName();
        this.point = p.getPoint();
        this.level = p.getLevel();
        this.created = LocalDate.now();
    }

    public String getName() {
        return name;
    }

    public int getPoint() {
        return point;
    }

    public int getLevel() {
        return level;
    }

    public LocalDate getCreated() {
        return created;
    }

    @Override
    public String toString() {
        return name + " " + created;
    }


    @Override
    public void load() {
        player.setLevel(level);
        player.setName(name);
        player.setPoint(point);
    }
}

File: Design Patterns/basic-patterns-java-kotlin/memento/player/PlayerMemento.kt
package memento.player

import java.time.LocalDate

class PlayerMemento(private val player: Player) : Memento {
    val name: String = player.name
    private val point: Int = player.point
    private val level: Int = player.level
    private val created: LocalDate = LocalDate.now()

    override fun toString(): String {
        return "$name $created"
    }

    override fun load() {
        player.level = level
        player.name = name
        player.point = point
    }

}